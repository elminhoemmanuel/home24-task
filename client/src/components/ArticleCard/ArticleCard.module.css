.article {
    border: 1px solid lavenderblush;
    padding: 10px;
  }
  
  .article > * {
    display: inline-block;
    padding: 4px 0;
    margin: 4px 0;
    width: 100%;
  }
  
  .article > section[role="button"] {
    padding: .2em;
    background-color: lightgoldenrodyellow;
    border: 1px solid lightgray;
    cursor: pointer;
    text-align: center;
  }