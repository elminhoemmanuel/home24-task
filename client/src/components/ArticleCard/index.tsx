import React from 'react';

import { Article } from '../../utils/types/productList';
import { formatter } from "../../utils/helpers/format";
import Pribtn from '../PriBtn';

const ArticleCard = ({ article }: { article: Article }) => {
    return (
        <div className='rounded border border-hmash hover:border-hmred p-4 cursor-pointer text-sm'>
            <div className='mb-3'><img className='h-52 md:h-48 w-full' src={article.images[0].path} alt="product image" /></div>
            <p data-testid="product-name">{article.name}</p>
            <p className='font-bold my-2'>{formatter.format(article.prices.regular.value / 100)}</p>
            <div className='mt-3 mb-2'>
                <Pribtn text="Add to cart" addStyle='block w-full' />
            </div>
        </div>
    )
};

export default ArticleCard