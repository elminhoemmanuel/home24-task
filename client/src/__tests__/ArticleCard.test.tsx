import React from 'react';
import { render, screen } from '@testing-library/react';
import { within } from '@testing-library/dom'
import ArticleCard from '../components/ArticleCard';

const article = {
    images: [{ path: "https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp" }],
    name: "Premium Komfortmatratze Smood",
    prices: { currency: "EUR", regular: { value: 54999 } },
    variantName: "180 x 200cm"
}


test("On render, check that ArticleCard is rendered with correct Image", () => {
    render(<ArticleCard article={article}/>);

    expect(screen.getByAltText("product image"))

});

test("On render, check that ArticleCard is rendered with correct product name", () => {
    const { getByTestId } = render(<ArticleCard article={article} />);
    const { getByText } = within(screen.getByTestId('product-name'))
    expect(getByText("Premium Komfortmatratze Smood")).toBeInTheDocument()

});

