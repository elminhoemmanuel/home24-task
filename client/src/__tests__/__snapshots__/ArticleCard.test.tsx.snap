// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that ArticleCard is rendered with correct Image 1`] = `
<div
  class="rounded border border-hmash hover:border-hmred p-4 cursor-pointer text-sm"
>
  <div
    class="mb-3"
  >
    <img
      alt="product image"
      class="h-52 md:h-48 w-full"
      src="https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp"
    />
  </div>
  <p>
    Premium Komfortmatratze Smood
  </p>
  <p
    class="font-bold my-2"
  >
    549,99 €
  </p>
  <div
    class="mt-3 mb-2"
  >
    <button
      class="text-sm rounded text-white p-3 cursor-pointer bg-hmred hover:bg-orange-500 block w-full"
    >
      Add to cart
    </button>
  </div>
</div>
`;
